### ISSUE TITLE
Smoke and explosions are wrong in resistence retribution 

### ISSUE THREAD

**Issue by antarusch** [hrydgard/ppsspp#10165]
Greetings,

I´m playing resistance retribution on the last development versions of ppsspp and the graphics for the smoke and the explosions are wrong, this happens in vulkan and opengl, also I´m using android.

Some examples of this can be seen on the first level intro of the game, when the protagonist is flying down in the parachute there are some explosions on the groud and smoke in the sky.

**Comment by hrydgard** [comment]
Screenshots please?

**Comment by antarusch** [comment]
Ok. here´s two screenshots, in one of the screenshots there´s an explosion but i don´t think is possible to capture the error in a screenshot.

In the screenshot with the fire shows the error with the smoke.
![screenshot_20171120-124158](https://user-images.githubusercontent.com/29181077/33030503-d4598756-cdf1-11e7-91d9-2b317ccd2c65.png)
![screenshot_20171120-124259](https://user-images.githubusercontent.com/29181077/33030512-d96a473a-cdf1-11e7-895a-b8cd41cc4f30.png)

**Comment by unknownbrackets** [comment]
Presumably the smoke should be more diffuse and not a thick cloud like that - right?  Does this happen at 1x render resolution?

Is it supposed to look like this?
https://youtu.be/qGFVfFwRJF8?t=1m39s
https://youtu.be/M4qYMAzgsjs?t=1m18s

I'm assuming since the latter video was released in 2009, it's probably right.  I don't have this game and haven't played it, though, so I can only guess at how it should look.

This could just be an effect that doesn't look great at higher resolutions...

-[Unknown]

**Comment by antarusch** [comment]
Yes, this happens at 1x render resolution, in the first video that you posted beginning at  2:40 min you can see the smoke being render correctly in the fire (the same area of the second screenshot that I posted).

**Comment by unknownbrackets** [comment]
Could you try exporting a GE debugger dump on PC?  Assuming this happens in the latest version, still.

To do this, open the game and select Debug -> GE debugger..., then when it's displaying the scene, press Record in the top right.  After a second or so, it'll finish and save a trace of the drawing activity.

After that, check the memstick/PSP/SYSTEM/DUMP folder and it'll have created a file named something like "ULES12345_0000.ppdmp".  You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by antarusch** [comment]
Hi,

Sorry for my late reply, my cel got damaged and until very recently I got a
new one,

I tried the last development build on android ( v1.7.1-243-g80e7f083e
<https://github.com/hrydgard/ppsspp/commit/80e7f083e3935033b017097ab9f6d912f0b6ff20>)
and the issues are still there, I can do the steps previously described in
android?

Best regards,

**Comment by unknownbrackets** [comment]
This describes how to do it on Android:
https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump

It wasn't supported before, but support was added recently.

-[Unknown]

**Comment by LunaMoo** [comment]
Seems like another game might be using the same engine with same smoke glitch as this and Siphon Filter ~  [Socom 3](https://forums.ppsspp.org/showthread.php?tid=25023).

**Comment by sifvodich** [comment]
Someone fix this issue, please!

**Comment by stg43x** [comment]
still need a fix!

**Comment by jimkage** [comment]
I think the smokes are still bugged 

https://files.catbox.moe/ofx913.mp4

**Comment by sifvodich** [comment]
This is an important game on PSP but this smoke graphic bug makes it unbearable to play, I wonder why this bug has not been fixed yet.
